src/server/scripts/Custom/functional_ai_tests.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+* Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <memory>
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+
+#include "FunctionalScheduler.h"
+#include "FunctionalCreatureAI.h"
+
+class npc_simple_scheduler_test : public CreatureScript
+{
+public:
+    npc_simple_scheduler_test() : CreatureScript("npc_simple_scheduler_test") { }
+
+    struct npc_simple_scheduler_testAI : public CreatureAI
+    {
+        FunctionScheduler scheduler;
+
+        uint32 step;
+
+        npc_simple_scheduler_testAI(Creature* creature) : CreatureAI(creature)
+        {
+        }
+
+        void Reset() override
+        {
+            // Cancel all tasks
+            scheduler.CancelAll();
+
+            step = 0;
+            me->SetReactState(REACT_AGGRESSIVE);
+        }
+
+        void EnterCombat(Unit* /*who*/) override
+        {
+            me->Yell("Mh who is that...", LANG_UNIVERSAL);
+
+            // Schedule a simple task that gets executed once in 5 seconds.
+            // As time unit std::chrono::duration is used which offers typedef for common used timeunits.
+            // No magic milliseconds anymore!
+            scheduler.Schedule(Seconds(5), [this](TaskContext& context)
```

Comment:
Couldn't this be written as `5s`? http://en.cppreference.com/w/cpp/chrono/operator%22%22s